docs/Bed_Mesh.md

```diff
@@ -370,14 +370,54 @@ are identified in green.
 
 ![bedmesh_interpolated](img/bedmesh_faulty_regions.svg)
 
+### Adaptive Meshes
+
+Adaptive bed meshing is a way to speed up the bed mesh generation by only probing
+the area of the bed used by the objects being printed. When used, the method will
+automatically adjust the mesh parameters based on the area occupied by the defined
+print objects.
+
```

Comment:
@kyleisah, I have added this to the documentation in a reworded/reorg'ed form.

@Arksine, I would appreciate a review of the new documentation section from you, as well?

Reply:
See the review point [here](https://github.com/Klipper3d/klipper/pull/6461#discussion_r1462263644).  Hopefully its clear enough, Github doesn't seem to allow suggestions outside of the modified code.  Essentially we just set `self._profile_name = None` right before returning `True` from `set_adaptive_mesh()`.  Then we modify the last line `probe_finalize()` to prevent from sending the `None` value `save_profile()`.